src/ixdat/calculators/ec_calculators.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Aug  8 13:29:09 2024
+
+@author: Søren
+"""
+from ..measurements import Calculator
+from ..data_series import ValueSeries
+from ..plotters.ec_plotter import EC_FANCY_NAMES
+from .scan_rate_tools import calc_sharp_v_scan
+
+
+class ECCalibration(Calculator):
+    """An electrochemical calibration with RE_vs_RHE, A_el, and/or R_Ohm"""
+
+    calculator_type = "EC calibration"
+    extra_column_attrs = {"ec_calibration": {"RE_vs_RHE", "A_el", "R_Ohm"}}
+    # TODO: https://github.com/ixdat/ixdat/pull/11#discussion_r677552828
+
+    def __init__(
+        self,
+        name=None,
+        technique="EC",
+        tstamp=None,
+        measurement=None,
+        RE_vs_RHE=None,
+        A_el=None,
+        R_Ohm=None,
+    ):
+        """Initiate a Calibration
+
+        Args:
+            name (str): The name of the calibration
+            technique (str): The technique of the calibration
+            tstamp (float): The time at which the calibration took place or is valid
+            measurement (ECMeasurement): Optional. A measurement to calibrate by default
+            RE_vs_RHE (float): The reference electrode potential on the RHE scale in [V]
+            A_el (float): The electrode area in [cm^2]
+            R_Ohm (float): The ohmic drop resistance in [Ohm]
+        """
+        super().__init__(
+            name=name, technique=technique, tstamp=tstamp, measurement=measurement
+        )
+        self.RE_vs_RHE = RE_vs_RHE
+        self.A_el = A_el
+        self.R_Ohm = R_Ohm
+
+    def __repr__(self):
+        # TODO: make __repr__'s consistent throught ixdat
+        return (
+            f"{self.__class__.__name__}"
+            f"(RE_vs_RHE={self.RE_vs_RHE}, A_el={self.A_el}, R_Ohm={self.R_Ohm})"
+        )
+
+    available_series_names = {"potential", "current"}
+    # This is a constant class attribute rather than a property because an
+    # ECCalibration always returns potential and current, as correctly as possible.
+
+    def calculate_series(self, key, measurement=None):
+        """Return a calibrated series for key based on the raw data in the measurement.
+
+        Key should be "potential" or "current". Anything else will return None.
```

Comment:
This is handled in `__getitem__`. The idea is that getitem should be able to loop through calculators and if one fails to give it the value, it goes on to the next. Slightly easier with a `return None` than with an error to try and except.

Reply:
I've now made it so that the `calculate_series` method of all `Calculator` classes raises a `QuantificationError` if it is asked to return a series that it can't. (Maybe `QuantificationError` should be renamed `CalculationError`?)